optuna/samplers/nsgaii/_crossovers/_vsbx.py

```diff
@@ -30,13 +36,45 @@ class VSBXCrossover(BaseCrossover):
             Distribution index. A small value of ``eta`` allows distant solutions
             to be selected as children solutions. If not specified, takes default
             value of ``2`` for single objective functions and ``20`` for multi objective.
+        uniform_crossover_prob:
+            ``uniform_crossover_prob`` is the probability of uniform crossover
+            between two individuals selected as candidate child individuals.
+            This argument is whether or not two individuals are
+            crossover to make one child individual.
+            If the ``uniform_crossover_prob`` exceeds 0.5,
+            the result is equivalent to ``1-uniform_crossover_prob``,
+            because it returns one of the two individuals of the crossover result.
+            If not specified, takes default value of ``0.5``.
+            The range of values is ``[0.0, 1.0]``.
+        use_child_gene_prob:
+            ``use_child_gene_prob`` is the probability of using the value of the generated
+            child variable rather than the value of the parent.
+            This probability is applied to each variable individually.
+            where ``1-use_chile_gene_prob`` is the probability of
+            using the parent's values as it is.
+            If not specified, takes default value of ``0.5``.
+            The range of values is ``(0.0, 1.0]``.
     """
 
     n_parents = 2
 
-    def __init__(self, eta: float | None = None) -> None:
+    def __init__(
+        self,
+        eta: float | None = None,
+        uniform_crossover_prob: float = 0.5,
+        use_child_gene_prob: float = 0.5,
+    ) -> None:
         self._eta = eta
```

Comment:
Is the valid range for $\eta$ is $\geq 0$? I want to validate the value here since I find that setting $\eta = -1$ causes ZeroDivisionError. This may also happen in SBXCrossOver.
```python
    beta_1 = np.power(1 / (2 * us), 1 / (eta + 1))
                                    ~~^~~~~~~~~~~
ZeroDivisionError: division by zero
```

Reply:
I could not find a description of the range of `eta` in the references listed in the document, but the following ones mention that it is a non-negative value.
Add a check for non-negative values, since there is also the issue of zero divisions.

[Self-adaptive simulated binary crossover for real-parameter optimization](https://dl.acm.org/doi/10.1145/1276958.1277190)

I will implement the same for the SBX as a another PR.